src/sys.rs

```diff
@@ -0,0 +1,425 @@
+
+pub use self::inner::*;
+
+#[cfg(unix)]
+mod inner {
+    use libc::{self, c_int, c_long, c_char, time_t};
+    use ::Tm;
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub use self::mac::*;
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    pub use self::unix::*;
+
+    /// ctime's `tm`
+    #[repr(C)]
+    struct tm {
+        tm_sec: c_int,
+        tm_min: c_int,
+        tm_hour: c_int,
+        tm_mday: c_int,
+        tm_mon: c_int,
+        tm_year: c_int,
+        tm_wday: c_int,
+        tm_yday: c_int,
+        tm_isdst: c_int,
+        tm_gmtoff: c_long,
+        tm_zone: *const c_char,
+    }
+
+    impl Default for tm {
+        fn default() -> tm {
+            tm {
+                tm_sec: 0,
+                tm_min: 0,
+                tm_hour: 0,
+                tm_mday: 0,
+                tm_mon: 0,
+                tm_year: 0,
+                tm_wday: 0,
+                tm_yday: 0,
+                tm_isdst: 0,
+                tm_gmtoff: 0,
+                tm_zone: 0 as *const c_char
+            }
+        }
+    }
+
+    impl<'a> From<&'a Tm> for tm {
```

Comment:
Same from this as above, I think this should be a function. Also, the old C version did not set gmtoff, which may be significant.